nexusparser/tools/yaml2nxdl/yaml2nxdl_recursive_build.py

```diff
@@ -8,17 +8,26 @@
 import os, sys
 import yaml
 from lxml import etree
-from yaml2nxdl_utils import nx_base_clss_string_mangling, nx_name_type_resolving
+from yaml2nxdl_utils import nx_name_type_resolving
 from yaml2nxdl_utils import nx_base_clss, nx_cand_clss, nx_unit_idnt, nx_unit_typs
 from yaml2nxdl_utils import nx_type_keys, nx_attr_idnt
 
 def xml_handle_docstring(obj, k, v):
+    """
+    Xml attribute on existing object node in an xml tree
+    """
     obj.set('doc', str(v))
 
 def xml_handle_units(obj, k, v):
+    """
+    Xml attribute on existing object node in an xml tree
+    """
     obj.set('units', v)
 
 def xml_handle_exists(obj, k, v):
+    """
+    Xml attribute on existing object node in an xml tree
+    """
     if v != None:
```

Comment:
Not part of the PR. v, k, vv, kk are terrible names that do not help in understanding whats happening.